patches/server/0850-Pose-Api.patch

```diff
@@ -0,0 +1,100 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: LoaiDev <lolyakram11@gmail.com>
+Date: Tue, 11 Jan 2022 18:02:25 +0200
+Subject: [PATCH] Pose Api
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
+index 8ca6388ddd834c4f0e94450c8a6ce6b5e73a5f60..e1e3790f831a2f4aca953c3e7d79a9b0cff01c84 100644
+--- a/src/main/java/net/minecraft/world/entity/Entity.java
++++ b/src/main/java/net/minecraft/world/entity/Entity.java
+@@ -326,6 +326,8 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
+     private org.bukkit.util.Vector origin;
+     @javax.annotation.Nullable
+     private UUID originWorld;
++    private boolean persistPose;
++    private boolean fixedPose;
+ 
+     public void setOrigin(@javax.annotation.Nonnull Location location) {
+         this.origin = location.toVector();
+@@ -613,6 +615,17 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
+     public void onClientRemoval() {}
+ 
+     public void setPose(net.minecraft.world.entity.Pose pose) {
++    // Paper start - Pose api
++        if (this.fixedPose) {
++            return;
++        }
++        setPose(pose, false, false);
++    }
++
++    public void setPose(net.minecraft.world.entity.Pose pose, boolean persist, boolean fixed) {
++        this.fixedPose = fixed;
++        this.persistPose = persist;
++        // Paper end
```

Comment:
Additionally here, I wouldn't update the method signature.
Instead, on CraftEntity#setPose directly access and set the fields yourself.

So it would look like
```java
public void setPose(net.minecraft.world.entity.Pose pose) {
       if (this.fixedPose) { 
           return;
        }
        // CraftBukkit start
        if (pose == this.getPose()) {
            return;
        }
        this.level.getCraftServer().getPluginManager().callEvent(new EntityPoseChangeEvent(this.getBukkitEntity(), Pose.values()[pose.ordinal()]));
        // CraftBukkit end
        this.entityData.set(Entity.DATA_POSE, pose);
    }
```

(api)

```java
     @Override
      public void setPose(Pose pose, boolean fixed) {
          CraftEntity handle = this.getHandle();
          handle.persist = true;
          handle.fixedPose = false;
          handle.setPose(net.minecraft.world.entity.Pose.values()[pose.ordinal()));
          handle.fixedPose = fixed;
     }
```

This change is a bit iffy, but the idea is you want to try to minimize the diff. (Although it might not matter that much)